### DISCUSSION TITLE
GraphQL: Filter for tags that have at least one entry associated

### DISCUSSION THREAD

**Discussion by julianklotz** [craftcms/cms#12134] [Q&A] [answered]
Hi there,

I’m just wondering how to filter for tags that have at least one entry associated.

Our use case:

* We list events on our website (events have their own section)
* Events are tagged by topic, e.g. with a movie, a band, …
* The events expire as soon as they’re over
* We’d like to show a tag bubble in the main menu (Show all events for the movie A, show all concerts of band B)
* The tag bubble should exclude tags that are not linked to active events  

In pseudo code: `tags = Tags.filter('entries__active'=True).annotate(entry_count=Count('entries__count')).filter(entry_count > 0)`

I haven't yet figured out how to accomplish this:

* `TagCount` doesn't seem to help
* The Tag’s `_count` field doesn't seem to help, because there’s no field to _query_  related entries (_filtering_ is possible, though)

Any thoughts, ideas?

Thanks!

**Comment by brandonkelly** [comment, marked as answer]
You can accomplish that using the `relatedToEntries` argument:

```graphql
query {
  tags(relatedToEntries:{}) {
    title
  }
}
```

(You can filter down _which_ entries the tags must be related to by passing nested arguments inside that `{}`, but as it stands that will return all tags that are related to _any_ entries.)

**Reply by julianklotz** [reply]
Thank you Brandon, lovely!
